src/Compilers/CSharp/Test/Semantic/Semantics/NullCheckedParameterTests.cs

```diff
@@ -114,25 +114,71 @@ public void NullCheckedPartialMethod()
             var source = @"
 partial class C
 {
-    partial void M(string x!);
+    partial void M(string x!!);
 }";
             CreateCompilation(source, parseOptions: TestOptions.RegularPreview).VerifyDiagnostics(
                     // (4,27): error CS8717: Parameter 'x' can only have exclamation-point null checking in implementation methods.
-                    //     partial void M(string x!);
+                    //     partial void M(string x!!);
                     Diagnostic(ErrorCode.ERR_MustNullCheckInImplementation, "x").WithArguments("x").WithLocation(4, 27));
         }
 
+        [Fact]
+        public void NullCheckedBadSyntax()
+        {
+            var source = @"
+partial class C
+{
+    void M0(string name !!=null) { }
+    void M1(string name! !=null) { }
+    void M2(string name!!= null) { }
+    void M3(string name ! !=null) { }
+    void M4(string name ! !=null) { }
+    void M5(string name! ! =null) { }
+}";
+            CreateCompilation(source, parseOptions: TestOptions.RegularPreview).VerifyDiagnostics(
+                    // (4,20): warning CS8893: Parameter 'name' is null-checked but is null by default.
```

Comment:
Consider either turning off this warning, or using a non-null default to reduce noise.

Reply:
Thanks for the tip, turned off that warning.